urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/agent/infocollect/DynamicNetworkInfo.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.text.DecimalFormat;
+import java.util.StringTokenizer;
+
+/**
+ * Created with IntelliJ IDEA. User: Administrator Date: 13-10-23 Time: 下午2:29
+ * To change this template use File | Settings | File Templates.
+ */
+public class DynamicNetworkInfo implements Runnable {
+
+	private static final Log LOG = LogFactory.getLog(DynamicNetworkInfo.class);
+
+	private int samplingInterval;
+
+	private volatile boolean running = true;
+
+	public DynamicNetworkInfo(int samplingInterval) {
+		this.samplingInterval = samplingInterval;
+	}
+
+	public void getNetworkInfoByProc() throws IOException {
+		double inReceivesPs = 0.00; // IP接受包率 包数/秒
+		double inDeliversPs = 0.00; // IP回应包率 包数/秒
+		double outRequestsPs = 0.00; // IP请求包率 包数/秒
+		double inSegsPs = 0.00; // TCP接受段率 段数/秒
+		double outSegsPs = 0.00; // TCP发送段率 段数/秒
+		double retransSegsPs = 0.00; // TCP重发段率 段数/秒
+		double inDatagramsPs = 0.00; // UDP接受包率 包数/秒
+		double outDatagramsPs = 0.00; // UDP发送包率 包数/秒
+		double[] bufferB = getCommondTCPUDPIP();
+		int data = 3;
+		try {
+			Thread.sleep(3000);
+		} catch (InterruptedException e) {
+			e.printStackTrace(); // To change body of catch statement use File |
+									// Settings | File Templates.
+		}
+		double[] bufferA = getCommondTCPUDPIP();
+		if ((bufferA != null) & (bufferB != null)) {
+			DecimalFormat df = new DecimalFormat("000.00");
+			inReceivesPs = Double.parseDouble(df.format((bufferA[0] - bufferB[0]) / data));
+			inDeliversPs = Double.parseDouble(df.format((bufferA[1] - bufferB[1]) / data));
+			outRequestsPs = Double.parseDouble(df.format((bufferA[2] - bufferB[2]) / data));
+			inSegsPs = Double.parseDouble(df.format((bufferA[3] - bufferB[3]) / data));
+			outSegsPs = Double.parseDouble(df.format((bufferA[4] - bufferB[4]) / data));
+			retransSegsPs = Double.parseDouble(df.format((bufferA[5] - bufferB[5]) / data));
+			inDatagramsPs = Double.parseDouble(df.format((bufferA[6] - bufferB[6]) / data));
+			outDatagramsPs = Double.parseDouble(df.format((bufferA[7] - bufferB[7]) / data));
+			SharedArea.dynamicInfoSerializable.setInReceivesPs(inReceivesPs);
+			SharedArea.dynamicInfoSerializable.setInDeliversPs(inDeliversPs);
+			SharedArea.dynamicInfoSerializable.setOutRequestsPs(outRequestsPs);
+			SharedArea.dynamicInfoSerializable.setInSegsPs(inSegsPs);
+			SharedArea.dynamicInfoSerializable.setOutSegsPs(outSegsPs);
+			SharedArea.dynamicInfoSerializable.setRetransSegsPs(retransSegsPs);
+			SharedArea.dynamicInfoSerializable.setInDatagramsPs(inDatagramsPs);
+			SharedArea.dynamicInfoSerializable.setOutDatagramsPs(outDatagramsPs);
+		}
+	}
+
+	/**
+	 * 从/proc/net/snmp中获取TCP/UDP/IP的信息
+	 * 
+	 * @return mind
+	 * @throws IOException
+	 */
+	private double[] getCommondTCPUDPIP() throws IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 27. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.